ml-algorithms/src/test/java/org/opensearch/ml/engine/algorithms/remote/HttpJsonConnectorExecutorTest.java

```diff
@@ -59,6 +64,124 @@ public void setUp() {
+                .build();
+            HttpJsonConnectorExecutor executor = spy(new HttpJsonConnectorExecutor(connector));
+            when(executor.getHttpClient()).thenReturn(httpClient);
+            MLInputDataset inputDataSet = TextDocsInputDataSet.builder().docs(Arrays.asList("test doc1", "test doc2")).build();
+            ModelTensorOutput modelTensorOutput = executor.executePredict(MLInput.builder().algorithm(FunctionName.REMOTE).inputDataset(inputDataSet).build());
+            verify(executor, Mockito.times(1)).invokeRemoteModel(any(), any(), any(), any());
+            Assert.assertEquals(1, modelTensorOutput.getMlModelOutputs().size());
+            Assert.assertEquals("response", modelTensorOutput.getMlModelOutputs().get(0).getMlModelTensors().get(0).getName());
+        } catch (Exception e) {
+            fail("An exception was thrown: " + e.getMessage());
+        }
+    }
+
+    public void invokeRemoteModel_HeaderTest(Map<String, String> header) throws ClientProtocolException, IOException {
+        ConnectorAction predictAction = ConnectorAction.builder()
+            .actionType(ConnectorAction.ActionType.PREDICT)
+            .method("GET")
+            .headers(header)
+            .url("http://test.com/mock")
+            .requestBody("{\"input\": ${parameters.input}}")
+            .build();
+        when(httpClient.execute(any())).thenReturn(response);
+        HttpEntity entity = new StringEntity("{\"response\": \"test result\"}");
+        when(response.getEntity()).thenReturn(entity);
+        Connector connector = spy(HttpConnector
+            .builder().name("test connector")
+            .version("1")
+            .protocol("http")
+            .actions(Arrays.asList(predictAction))
+            .build());
+        HttpJsonConnectorExecutor executor = spy(new HttpJsonConnectorExecutor(connector));
+        when(executor.getHttpClient()).thenReturn(httpClient);
+        MLInputDataset inputDataSet = TextDocsInputDataSet.builder().docs(Arrays.asList("test doc1", "test doc2")).build();
+        ModelTensorOutput modelTensorOutput = executor.executePredict(
+            MLInput.builder().algorithm(FunctionName.REMOTE)
+            .inputDataset(inputDataSet)
+            .build()
+        );
+        // verify(connector, Mockito.times(1)).getDecryptedHeaders();
+        Assert.assertEquals(header, executor.getConnector().getDecryptedHeaders());
+    }
+    @Test
+    public void invokeRemoteModel_GETMethodSuccessPath() {
+        invokeRemoteModel_SuccessPath("GET");
+    }
+    @Test
+    public void invokeRemoteModel_POSTMethodSuccessPath() {
+        invokeRemoteModel_SuccessPath("POST");
+    }
+
+    @Test
+    public void invokeRemoteModel_HeaderNull() throws ClientProtocolException, IOException {
+        invokeRemoteModel_HeaderTest(null);
+    }
+
+    @Test
+    public void invokeRemoteModel_HeaderNotNull() throws ClientProtocolException, IOException {
```

Comment:
Looks like this test is failing. Could you please check?

Reply:
Do you have an IDE, e.g. Intellij, set up to debug this?

Reply:
@austintlee I solved this issue for him.